packages/react-native/React/CoreModules/RCTAppearance.mm

```diff
@@ -71,6 +67,19 @@ @implementation RCTAppearance {
   NSString *_currentColorScheme;
 }
 
+- (instancetype)init
+{
+  if ((self = [super init])) {
+    UITraitCollection *traitCollection = RCTSharedApplication().delegate.window.traitCollection;
+    _currentColorScheme = RCTColorSchemePreference(traitCollection);
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(appearanceChanged:)
+                                                 name:RCTUserInterfaceStyleDidChangeNotification
+                                               object:nil];
+  }
```

Comment:
You should move the logic from `stopObserving` to `invalidate` since we always want to keep the listener.

Reply:
Done